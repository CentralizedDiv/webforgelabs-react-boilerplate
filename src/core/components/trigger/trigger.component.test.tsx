import React from 'react';
import STTrigger from './trigger.component';
import { shallow } from 'enzyme';

describe('STTrigger', () => {
  const mockOnClick = jest.fn();

  const collapsed = true;
  const component = shallow(<STTrigger onClick={mockOnClick} collapsed={collapsed}/>);

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call onClick method when clicked', () => {
    component.simulate('click');
    expect(mockOnClick).toHaveBeenCalled();
  });

});