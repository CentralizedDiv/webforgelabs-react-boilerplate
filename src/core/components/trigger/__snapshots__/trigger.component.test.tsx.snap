// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STTrigger renders properly 1`] = `
<div
  className="st-trigger st-trigger--collapsed"
  onClick={[MockFunction]}
>
  <Icon
    type="menu"
  />
  <span>
    layout.expand_sider
  </span>
</div>
`;
